### ISSUE TITLE
Applied Energistics - Vanishing Items/NBTTagData

### ISSUE THREAD

**Issue by ray73864** [MinimallyCorrect/TickThreading#246]
Possible causes:
- Inventory concurrency issue - likely, as it happens sporadically
- Bug in new chunk provider - not likely, as the issue can occur without reloading the chunk.

**Comment by LunNova** [comment]
Original text:

> According to the FAQ on ae-mod.info (http://ae-mod.info/Frequently-Asked-Questions/) , it has:
> 
> > All my Drives Wiped Clean, or All my Storage Cells are gone
> > 
> > This is probobly related to another mod called TickThreading, which can cause issues with Saving Chunk Data the end >result of these tends to be these issues.
> 
> Is this still an issue? i have a lot of stuff in my ME storage network, 6 storage cells so far, i have a friend who has 2 ME drives both full of storage cells.
> 
> So far i am liking TT, i can have 700 mobs in my zombie grinder and still have a TPS of 20, where as before, 500mobs would drop the TPS down to 5.

Yes, this is probably still an issue. Although the cause listed by the FAQ is an incorrect simplification, as it happens while the chunk is loaded, no unload/reload cycle is necessary. I haven't had time to fix it.

**Comment by utoxin** [comment]
Interested in this bug being fixed as well. Would love to use TT with the Ultimate FTB modpack. :)

**Comment by aquagene** [comment]
Interested as well now that DW20 has added Applied Energistics. It's popular enough that I'm forced to not use Tickthreading for fear people will lose items on my server.

**Comment by nylser** [comment]
Would also be very interested in this bug to be fixed, I'm running an FTB Ultimate Server on an 8-core server, which is getting laggy pretty fast with standard forge server, but with TickThreading its an awesome performance and balances everything over the 8 cores! Keep up the good work, Nallar!

**Comment by Delaura** [comment]
So there is a confirmed bug between AE and TT that makes items dissapear? So I'm not crazy when I say that 5937 fertilizer went missing in my AE system? All my fermenters were located in a single chunk, and once I hooked up my  AE System, it worked fine, until I logged back in 2 hours later, when it was all gone, and just the saplings remained, which were stored in a deep storage unit in another chunk.

**Comment by HaoSs07** [comment]
just happen to me .... nasty bug, probably just have to remove TT till a fix is made, a lot of users started to use ME system

**Comment by xfloh22** [comment]
has this bug been fixed with the other chunk saving fixes?

**Comment by LunNova** [comment]
I also recently fixed some AE issues which may be related. Might be fixed, not sure.

**Comment by xfloh22** [comment]
k il test it in a sec

**Comment by HaoSs07** [comment]
was working for 3 days with no problems,but now after the deadlock bug fix, i lost 3 times data from disk in under 1  hour , don't know if its really related or i just have bad luck, xfloh can you test this with last build ?

**Comment by xfloh22** [comment]
testing it atm with completely new versíon of everything. fresh map as well, tell you how it goes

**Comment by xfloh22** [comment]
odd, i havent had any problems with dissapearing items yet. i let the chunks unload for about 15 minutes and then came back. i also logged out and back in and the items were still there

**Comment by HaoSs07** [comment]
get a query and a ender chest and fill some disks

**Comment by HaoSs07** [comment]
will update and test

**Comment by xfloh22** [comment]
items just dissapeared in chunk that gets unloaded. items stayed in the one that was allways loaded

**Comment by alicheek** [comment]
I think thats meant to happen if u have cleanup on or what ever that config option is on TT

**Comment by xfloh22** [comment]
first of all i have that off and second that would be a terrible feature.. im not sure it is meant to erase AE data...

**Comment by LunNova** [comment]
Cleanup is to reduce the memory leaked if a mod holds onto world objects.

**Comment by xfloh22** [comment]
well yes but having cleanup erase AE data is a terrible feature

**Comment by LunNova** [comment]
I never said it did, and it doesn't. Something else is causing this. Intentionally erasing AE data would be stupid :P

**Comment by HaoSs07** [comment]
still happens but more rare, so strange, i'm lucking at the driver and it jumps from 0 to 63 types, then back to 0

**Comment by geoffcorey** [comment]
Users are starting to use AE and we have build 1308 (after the possible fix in build 1306) and we had issues last night with items disappearing.

**Comment by geoffcorey** [comment]
Another admin looked over the AE system that was losing items.   His response was, "I'm not positive it was due to TT. The power going to his ayeazall's AE system was through powerconverters mod and seemed VERY sketchy at best. He hooked it up directly to a power source and it seemed fine after that."

**Comment by HaoSs07** [comment]
ftb ultimate 1.0.2 is under testing, (latest 1.4.7 ) that includes some AE fixes ,
also eloraam twitted about an hour ago, and it seems she is back on RP2 for 1.5 so we may see a update to 1.5 directly (more updates to AE), the best thing to do right now is probably to wait and test with the updated version, also i formated the disks for testing, and so far in 2 days no items was lost, probably just luck ?

**Comment by utoxin** [comment]
An AE system doesn't lose items when it loses power. It can't accept /new/ items, but it won't lose the old ones.

**Comment by HaoSs07** [comment]
i lost items where the power is not enough, it starts/stops many times and for some reason wiped some disks ,

now i test with formatted disks and it seems to work

**Comment by ray73864** [comment]
If the power isn't enough, what i do, is i stick a redstone energy cell below the controller and power the cell.

Set the cell to output just above what your controller is asking for, basically a UPS for your network :)

**Comment by LunNova** [comment]
Can I call this a feature then? ;)

**Comment by ray73864** [comment]
Well no, because my energy cell is currently set at 70MJ/t, which means i have 30MJ/t remaining and my network keeps on expanding, just like the universe _cue the universe song from Monty Python and the Meaning of Life_.

And i doubt it is a power issue causing the problem, but then again, AlgorithmX2 has made some changes in his latest versions for 1.5, so maybe he fixed the problem, no idea.

**Comment by xfloh22** [comment]
well the formatting disc thing might be the reason i wasnt loosing any items... i allways format my discs.

**Comment by HaoSs07** [comment]
there is no point to set the cell, it will only use what it needs, i also use MJ to power it, and i have connected to the same grid: AE system , 4 quarry's that run full speed , and the excess MJ is connected to a HV EU transformer that goes in to matterfab,

**Comment by ray73864** [comment]
Except that 4 quarries don't demand much in the way of power.

I have 3 36HP liquid fueled boilers, they output quite a decent amount of MJ, but as soon as i use my assembly table with over 90 lasers on it, the AE network starts blinking on and off due to unstable power.

And i like to set an output on the cell, it's my choice, and it allows me to keep an eye on how much power it is using without having to do the math of MJ to unit.

**Comment by HaoSs07** [comment]
you will have to set it highter anyway as AE uses different power levels for different operations 

and a full speed quarry eats like 80 MJ if i remember right

**Comment by ray73864** [comment]
Actually, the AE in FTB Ultimate doesn't use much power at all for different operations, what it says it requires on the controller is basically what it uses.

The AE for 1.5 however works very differently, it uses more power if you have the export bus set to 'stack' instead of 'single' for instance.

And a fullspeed quarry doesn't go any faster when you give it more than 75MJ/t. It uses something like 35MJ/t just to power the quarry at max speed and 60MJ/t for each block it breaks (or something like that.

**Comment by HaoSs07** [comment]
I'm at 215 energy units/t , if i craft, it goes higher ,  u can go test, it does not use power as you say, for the stack or single option but it does use more if you transfer items; we can debate this forever, lets stay on topic; data lose

**Comment by wormzjl** [comment]
A formatted disk DO lose items.
Tried to put some chunkloaders beside the disk but failed.

**Comment by alicheek** [comment]
Just to say on build  TT 1388 and one of my players put some of his items in those computer storage devices then went some where which unloaded the chunk, when he came back he had none of the items in there any more.

**Comment by wormzjl** [comment]
Is there anyway that we can evade this problem rightnow?

**Comment by LunNova** [comment]
No, sorry :( Will hopefully have time to try to find what's causing this today.

**Comment by floh22** [comment]
would be really awesome if this could get fixed. il help in any way i can if needed.

**Comment by HaoSs07** [comment]
+1, will help with any test you need

**Comment by LunNova** [comment]
What I need to fix this more hours in a day or another dev to work on the project. :p

Higher exams in a few weeks, don't have enough time to fix all these issues :(

**Comment by HaoSs07** [comment]
let me guess , no dev lying around with no job

**Comment by utoxin** [comment]
I have no mod experience (yet), or I would absolutely volunteer to help.

**Comment by alicheek** [comment]
Doing a great job nallar ! Dont let it get in the way of ur studies you have given us a working mod so dont worry they can always be fixed later! :)

**Comment by HaoSs07** [comment]
well this one is allready in the " later " stage :p, as its a grate mod and more users start to learn it,  wiping items hurts servers worldwide

**Comment by dantherrien** [comment]
Just had this happen again last night.  It has ONLY happened to me with ME chests, but multiple times.  Usually with disks that have a large number of items on it (over 100K dirt and last night over 130K charcoal)  I noticed a while back that the ME chest seemed to be causing problems as ./profile or ./ticks reported it as an offending block.  Not sure how different the code is between the drives and chests but this bug seems to be limited to chests for me.  Will try to reproduce tonight with items I dont' care about.

**Comment by HaoSs07** [comment]
i use only drives, and happens allot: but in lats days only to the first disk :) dont know why , also making sure chunk are always loaded seems to help

**Comment by floh22** [comment]
any progress? i know you dont have much time but any updates would be awesome :smile:

**Comment by LunNova** [comment]
If anyone's able to upload a compressed FTB ultimate world with a system which reliably has this issue (and give coords of where it is) that would be very helpful.

**Comment by HaoSs07** [comment]
i can give you the region for my house
https://dl.dropboxusercontent.com/u/76651664/r.-6.-5.mca  put in region folder in world (any world under ftp ultimate)
coordonates -2794 63 -2532
remove chunk loaders. enter exit the chunk many times, bug should happen

**Comment by Stoo42** [comment]
Just to add my voice - I also have had this issue on FTB Ultimate 1.01. The chunk was always loaded with a chickenchunks loader, and also appeared to only happen to the chests. I have an ME Drive nearly full with preformatted storage, and it wasn't touched.
The chests I had next to it (4 of them), had 1k, 4k, 4k, 64k storage. I could watch the chest interface and see the items slowly drain out. It seems they gradually "leak" rather than all being deleted at once.

I'm not sure the ME Drive is a red herring, as these storage were also formatted, so it might be that instead of the fact its a drive...

Definitely not an easy one I dont think, I dont envy you, nallar!

**Comment by HaoSs07** [comment]
i use world anchor, and it seems to help, chuckenchunks does not

**Comment by dantherrien** [comment]
My issue was always an immediate storage cell wipe when moving it into or out of an ME chest.  It only seemed to happen when I did it in quick succession but that might just be in my head.  I usually use the ME chest to see what is actually on a single disk outside of the regular terminal interface.  Wiped at least 2 disks on me so I have since discontinued using it and just gone the route of IO ports + preformatting to guarantee the contents of a disk.

**Comment by Stoo42** [comment]
I haven't tried world anchor, but when you need more than the 9 chunks it provides, its a bit of a pain having to have a separate system just to keep them stocked with e-pearls!

Also noticed that pulling storage cells in and out of chests seemed to speed up the loss process. Have to admit I didn't play with it much more than this before rolling server back to the backup I took before putting TT on....

@nallar if you need some more scenario testing, I have can fire the TT world back up and try stuff for you if it helps resolve the issue, or even if you need another set of eyes cast over the code in question, I've done a bit of threaded dev before...

**Comment by Maximuscr31** [comment]
I had too many issues with the me and chunkloader crashes. when TT worked my server wouldn't drop below 20 for anything hardly. When it broke though it was a constant struggle. I am thankful for the mod and think it will be fantastic eventually. I can't begin to fathom the dedication it takes to redo everything mojang should be doing themselves. I can't wait till it gets stable but it is too alpha for me right now.

**Comment by geoffcorey** [comment]
Currently running 1465 and several of us have been experiencing items disappear from ME.   I tried various things to try and replicate including loading items in and bouncing around worlds.   Flooding the system.   Positive it is not a power issue.   However there is one thing that seems to be constant when items are disappearing.   

When items are crafting and I go to another world (there is a chunk loader on the AE system) that is when things typically disappear and usually items that are used in the crafting recipe.   For instance I have a recipe for Ultimate Hybrid and it was waiting on uu.   Well then I fired up a recipe to make 1024 stone bricks from cobble and then bounced to another world.    Glowstone, Ender pearls and blaze rods went to zero.   Both the ender farm and blaze farm were running.

**Comment by geoffcorey** [comment]
Let me know if you want the backup of the world.   it currently is about 2G.   We can setup a chat session and I can walk you though it when you have time

**Comment by HaoSs07** [comment]
will test after we fix the teleport bug, trying to get a crash log now

**Comment by geoffcorey** [comment]
Able to reproduce with build 1481 last night.   

Pretty sure autocrafting is part of the equation that is causing the problem.   I am going to try and isolate the conditions in which this occurs.

**Comment by HaoSs07** [comment]
i lost items from a system that was not linked to any autocraft

**Comment by geoffcorey** [comment]
HaoSs07 do you have a lot import buses currently doing work when you lose work?   It is one of the scenarios I'm going to test today.

**Comment by geoffcorey** [comment]
ok, autocrafting computer/storage was taken offline.  I have a mob trap that does obsidian pipes to iron chest.  Gates pulsing the obsidian pipes and the iron chests have import buses.  I fired up all 3 spawners (witch, ender, wither skellie) and watched the inventory and right before my eyes I went from 500+ ender pearls to 1.

**Comment by geoffcorey** [comment]
So with the blaze rods and ender pearls I have a system with an emitter that will turn on export of pearls to an autocrafting table to produce ender eyes and blaze powder if it falls below a stack in the AE system.   I watched the inventory with mob farms in full swing and boom all my eyes disappeared as well as blaze rods.

I then removed the export buses from the autocrafting tables and let the system run and nothing disappeared.  I went back and added the export buses and so far nothing has disappeared.  

I do have a new theory.   If exporting to something that does not have room, such as an autocrafting table, the system may still export items but the recipient of the item cannot except so it is lost.   I suppose this could be true for machines with relays for autoprocessing ore.

However the outlier to that scenario is the fact that other items would get lost traversing the AE system.  I wonder if there is some limit on the number of items traversing AE and if it goes above that number those items get lost?

**Comment by HaoSs07** [comment]
well export bus has slots for exported certine items.
i had only a disk with a chest and come cells, deuterium, trimium, something like that, i exported empty cells and deuterium, but not the tritium, no export bus for that, but i lost the tritium too. i only had left some empty cells. 10% of what i have

**Comment by LunNova** [comment]
@Stoo42 

It would be very helpful if you could look into it, I just don't have the time to go through AE's code and find what's broken. I'm afraid I don't really have any idea of what's broken, so good luck if you do try :P

Thanks.

**Comment by Stoo42** [comment]
When I have observed this bug, it has drained ALL items on storage, even ones that had not been exported or imported or specified in any crafting.

its weird, import/export activity may be the trigger, but this may just be coincidence as most people with AE will be doing import/export constantly anyway. 

Its almost like the storage chips themselves are ticking. I'd guess the contents of storage chips are stored in Tag Data. Because _most_ items tag data size is in the order of a few bytes, and AE seems to be the first mod that will store up to 64kB in a single items tag data.... Maybe this is where we should be looking. Thats a lot of data in that hash table, if anything TT is doing to it to manage it doesn't take into account that the storage chips have such big tag data, then this may just be classic memory leak and not related to threading as such at all. Copy even 1k storage chips tag data to mem, but when you save it back for some reason its not saving the full thing back. In pretty much all other items in game, I cant think of a case where tag data for a single item would get this big.

This is a guess, but it accounts for why AE uses those storage sizes, and how it calculates how much is used. Its probably just measuring the size of that region of tag data for the storage chip item - I have to emphasize I haven't seen the code, this is just speculation from what I know of how items work.

If this is the case though, question is then why storage chips are ticking in the first place, or even touched by TT if they are just plain "items" - unless I have a fundamental misunderstanding of what ticking entities are in game, or what bits TT touches... :-P

**Comment by geoffcorey** [comment]
@Stoo42 you may be on to something there.   I plan on pulling the world down and see if I can replicate without TT.  If I can then I will try to upgrade Forge and AE on the Universal 1.0.1 and I did more testing this afternoon and it does seem to be a race condition and likely has something to do on the emitters.   It appears to be some sort of race condition.

I found this on AE defect list that seems like a candidate that might be related http://ae-mod.info/Tracker/0076/

**Comment by LunNova** [comment]
I don't change the storage of tag data, other than from a hash map to a
concurrent hash map. The only difference there is that it doesn't allow
null keys.

I expect the issue is when two AE blocks on the same network access
inventory at the same time. An easy way to confirm that it is due to
concurrent ticking of tile entities in one network would be to make an AE
network entirely contained within a single chunk, and not on any of the
chunk's borders. If the issue no longer occurs, then that's the issue, as
two tile entities in the same chunk are never ticked concurrently.

**Comment by HaoSs07** [comment]
you guys also should check why there is no wipe when the chunk where the disks are is loaded with worldanchor . Or it happens rare,but only to the first disk ( in my case )

**Comment by geoffcorey** [comment]
@nallar that sounds consistent with what I am observing.  When multiple things are running it would sometimes happen.   If nothing but one thing was running it would not.    I will redo my setup tonight and try it all in 1 chunk and see if there are any glitches or missing items.

**Comment by geoffcorey** [comment]
It would appear that putting the AE system in a single chunk has fixed the issue. I will continue to test the single chunk theory this weekend.

**Comment by geoffcorey** [comment]
So odd item with the test.  I did this single chunk in a mystcraft age.  The age is stable.   In the chunk I have a power cable running the inside edge of the the chunk to the AE system.   In the last few hours that entire glass fibre line went missing.  I dug up the floor and not just a segment but the whole line from the corner of the chunk to halfway across the chunk including the elbow that would take 2nd block from the chunk boundary.

UPDATE: Upon further investigation there was a server crash related to someone writing bad Lua code for their turtle.  Tough to pin this down.   TT errors logged do not seem to correlate to this AE test system and those defect tracked in other issues.

**Comment by geoffcorey** [comment]
1513 able to reproduce the problem (baseline)
1514 able to reproduce the problem
1515 able to reproduce the problem + able to reproduce the problem of the glass fiber line connected from MFSU to AE disappearing

**Comment by geoffcorey** [comment]
Ok build 1520 it was looking real promising.  No issues until I triggered an emitter on an autocrafting table in another chunk to keep 64 ender eyes and 64 blaze powder in stock.  Once that happened the stuff in transient went to zero - uu, ender pearls, blaze rods.  All gone.

**Comment by HaoSs07** [comment]
this problem is really annoying

**Comment by geoffcorey** [comment]
well I will say the work around of having the AE system in a single chunk works.  Only issue I have had is the occasional fibre optical cable disappearing from mfsu to the AE system.

**Comment by floh22** [comment]
i know what you mean. i cant use TT on my server because of this

**Comment by wormzjl** [comment]
For the moment, use storage bus + diamond chests.

**Comment by HaoSs07** [comment]
storage bus + chest/barrel works. but the it lags the server a little when you craft hard items

**Comment by floh22** [comment]
well im cant really stop people from using those items from ae. i really dont wanna ban em either

**Comment by geoffcorey** [comment]
I'm running TT and we all moved our systems to a single chunk without issues.

**Comment by Maximuscr31** [comment]
Any update on build 1520?

**Comment by HaoSs07** [comment]
geof already posted about 1520
" Ok build 1520 it was looking real promising. No issues until I triggered an emitter on an autocrafting table in another chunk to keep 64 ender eyes and 64 blaze powder in stock. Once that happened the stuff in transient went to zero - uu, ender pearls, blaze rods. All gone. "

**Comment by Maximuscr31** [comment]
I saw that but was wondering if there was any more updated info to it besides it just dumping items with autocrafting tables. That can easily be warned against if that is the only issue left with it. I just don't go through another restore session with my world if I don't have to if I try it again and ME chest/drives emptying would cause me not to try it again.

**Comment by LunNova** [comment]
I can't guarantee that that is the only possible issue after that partial/failed fix.

**Comment by dantherrien** [comment]
Geoff and I have rebuilt both of our systems into a single chunk each and have seen no further problems.  The issue was not only with autocrafting tables...seemed to be concurrent access on any inventory item across chunk boundaries that would wipe the item from your system.

**Comment by Maximuscr31** [comment]
That is what I was trying to understand. Thank you for explaining where it is at dantherrien.

**Comment by HaoSs07** [comment]
@nallar probably silly question., but have you talked with AE dev ? he sure knows his mod and may have useful tips, don't see why he wont help

**Comment by LunNova** [comment]
I have talked, but the version of AE you're on is very out of date so it doesn't help much.

The code in AE related to this has been redesigned in newer versions to improve performance, and the bug probably doesn't exist in newer AE versions anyway.

Even if it still happens in newer AE this would be a lot easier to fix, as the inventory code for AE was also cleaned up. It's already a pain understanding decompiled code when the decompiler gets it wrong, and when you can't remap it easily, and design issues in the version of AE used in FTB just make this worse.

**Comment by HaoSs07** [comment]
then probably is time to let it go and think of the 1.6 update, ftb  update to 1.5 will be private

**Comment by HaoSs07** [comment]
at first look it seems to be fixed :D but will need more tests before you can safely close this

**Comment by Maximuscr31** [comment]
Now this is exciting news. Thank you Nallar.

**Comment by floh22** [comment]
hows it looking on your end haos? i havent got any problems yet

**Comment by HaoSs07** [comment]
good so far. would be nice if  @geoffcorey can test too

**Comment by floh22** [comment]
well ima just add it to my server. lets see if any1 complains :)

**Comment by ray73864** [comment]
So if i stick this in, i won't be required to put all my AE stuff into a single chunk will i? because there is no way i could fit my entire AE automation stuff into 1 chunk.

**Comment by HaoSs07** [comment]
i have my system in 4 chunks

**Comment by ray73864** [comment]
And it will work on the current FTB Ultimate pack for 1.4.7?

**Comment by HaoSs07** [comment]
yes. i use ultimate 1.4.7

**Comment by utoxin** [comment]
Very excited to hear this. I'll be installing it... tomorrow, probably, barring any reports of issues.

**Comment by cjm721** [comment]
Ok I tested this some more. (Direwolf20 pack, kinda use that as my test before throwing on ultimate). Ways I tested:

Make several infinite loops using both export > chest > import across several chunks lines. Also export > Item tesseract > tesseract > me interface

With that above tested having parts of it chunk loaded and others not, as well as none loaded while the loops where going.

So to me this seems fixed

**Comment by HaoSs07** [comment]
after future testing. There seems to be a problem with importing/exporting cells. they vanish :( lost like 5k empty cells and all helium cells it crafted
Only happen when used industrial centrifuge,crafted berylium using electrolizer and no problems there... Preformating a disk just for cells seems to help.

There is good change with the last fix. but it seems is not enough.
Also a player has reported missing stacks of string from a disk

**Comment by ray73864** [comment]
bugger, and i was looking at putting TT back in, guess i'll just have to wait some more. Kinda of at a stopping point at present anyway, the uncertainty with the current packs because of RP2.

**Comment by HaoSs07** [comment]
well wait for others to confirm that there is a problem, we may have same mods, but every server works in his own twisted way :P

**Comment by HaoSs07** [comment]
so far only reported vanish is when you use the autocraft !, no vanish if you craft manually

**Comment by geoffcorey** [comment]
I will try and migrate the world to a higher version then 1.0.1 Ultimate and retest this weekend.

**Comment by floh22** [comment]
my server has not had any problems and people are constantly autocrafting

**Comment by HaoSs07** [comment]
will reset TT config as i have some experimental things on, and will retest

**Comment by LunNova** [comment]
Concurrent network ticks being enabled is likely to result in this issue,
so that could be it.

**Comment by HaoSs07** [comment]
that was off, but reset'ed anyway.
It seems i'm not crazy. ill have a video in 1 hours or so

**Comment by Maximuscr31** [comment]
So far I have had two guys have some missing items. They are also the ones who have the biggest bases/setups. No one else seems to have any issues.

**Comment by wormzjl** [comment]
ME crafting is behaving a bit strange for me, many of my tasks got cancelled immediately after giving order for no reason.

**Comment by HaoSs07** [comment]
http://www.youtube.com/watch?v=IIxQGpOPz5g

**Comment by LunNova** [comment]
> This video is private

**Comment by HaoSs07** [comment]
ups. fixed

also i tried to craft it with the terminal allways open. same result ( if somone thinks items vanished because i went to check the boilers)

**Comment by LunNova** [comment]
It's now only 11 days until my English exam and I really don't have the time to debug complex concurrency issues in other people's code.

I may get it fixed, but will prioritise issues which are easier to fix, so chances are it won't be fixed soon.

**Comment by furlmc** [comment]
Adding another video related to this issue:  http://youtu.be/KROCRefy1XY

Basically, ME Assembler multiblock pulling directly from disks results in too few of the crafting prerequisite items being pulled from the disks.  This is with 1589, Ultimate 1.1.0 (which is the latest r.9 AppEng version; 1.0.1 is r.9b I think).

**Comment by HaoSs07** [comment]
probably the only update that has any fixes for our problem is r.10+

**Comment by furlmc** [comment]
I think this is fixable with some attention to appeng's inventory for the assembler and concurrency.  I'm not familiar with TT at all so I'm not in a position to write a patch (probably also not in a position to be making these statements) but that's my guess.

Suppose I'll add a visualvm screenshot of sampler while running 100x Coil production with appeng filter, just for reference: http://i.imgur.com/CpOPd49.png

**Comment by geoffcorey** [comment]
Further testing and the good news and not so bad news.

I took 1.0.1 Ultimate and upgrade AE to rv9-i and Mystcraft to 10.1.    After starting the world and making sure there were no glaring errors I added 1594 build of Tick Threading.

Fired up the mob grinders:
- Ender
- Blaze
- Witch
- Skelli's

At the same time:
- Autocrafted 64 Utlimates
- Autocrafted 64 Electrolyzers
- Used emitters to keep stock of ender eye dust
- Matfab generating UU

I left the world and messed around in a mystcraft age for 5 minutes.

Came back and did not lose a single item and everything was still chugging along autocrafting.   Only error in TT was unrelated to the AE issue.

2013-05-11 11:59:18 [SEVERE] [TickThreading] Exception ticking world Overworld/0
java.lang.ArrayIndexOutOfBoundsException: 0
    at me.nallar.tickthreading.collections.LongList.add(LongList.java:25)
    at yp.spawnMobsQuickly(SpawnerAnimals.java:138)
    at in.b_net_minecraft_world_WorldServer_tick_p0(WorldServer.java:380)
    at in.b(WorldServer.java)
    at net.minecraft.server.MinecraftServer.doWorldTick(MinecraftServer.java:485)
    at me.nallar.patched.server.PatchMinecraftServer$TickRunnable.run(PatchMinecraftServer.java:609)
    at me.nallar.tickthreading.minecraft.ThreadManager$1.run(ThreadManager.java:48)
    at me.nallar.tickthreading.util.FakeServerThread.run(FakeServerThread.java:16)

**Comment by LunNova** [comment]
Anyone who was still experiencing item loss, can you retry with #1598 or later?

**Comment by geoffcorey** [comment]
So is it your understanding that #1598 will fix the problem without updating AE?

**Comment by HaoSs07** [comment]
was a bug in patching  AE. im testing right now, same environment as in video

also from r9b to r9i , i does see much updates retaled to this problem, anyway Monday ultimate 1.0.2 should be pushed to recommended build and we all update to latest 1.4.7 AE

**Comment by geoffcorey** [comment]
HaoSs07.   Did you test rv9-i on 1.0.1?   I could not reproduce in that setup.

**Comment by HaoSs07** [comment]
cant right now. server is public,its week end. will test next week if ftb don't update monday

**Comment by Meatnog** [comment]
We are on TT#1603 and MCPC+ L27, but still on FTB 1.0.1, item loss is still taking place even if everything is in 1 chunk.

**Comment by Nohau** [comment]
+1 tried with L27 and L22

**Comment by ray73864** [comment]
Perhaps this is now an issue with MCPC+, i don't use that, so assuming the issue is now only present on MCPC+ i should be good to go.

**Comment by geoffcorey** [comment]
I have had zero issues with a heavily used AE system in a single chunk with Ultimate 1.0.1 and TT.  We do not use any other add-ons such as MCPC+

**Comment by Maximuscr31** [comment]
Meatnog still lost items in a single chunk on tt1610 l27 with unformatted disks. He formatted and named the disks and has lost 0 items. This has been tested back to back with replication.

**Comment by Meatnog** [comment]
Bad news.   Tt1610 I27, with FTB 1.1.0  (AE r9i) is not fixed.  I just watched, realtime, my ME system eat 10,000 cobble in under 15 seconds.   If I was not actively watching it, I might have missed it.    I note this is with an ME system all in 1 chunk, using named storage disks (but not preformatted to only accept certain items).

The only test left is to test it with preformatted disks only accepting certain items, but I do not have high hopes.

**Comment by wormzjl** [comment]
Are you using ME chest or ME drive?

**Comment by Meatnog** [comment]
ME Drive with 64K storage units.

**Comment by Meatnog** [comment]
It is possible my new issue with item loss may be due to a bugged ME Level Emitter being stuck on, despite the graphics not showing that it is.     I'm doing yet-more-testing now.

Basically I had an ME Level Emitter pointed at an ME Export Bus pointed at an Incinerator.   The whole thing set to send cobble to the incinerator whenever the cobble level got over 10000 cobble.    It was working beautifully, but if the above situation happened (bugged level emitter stuck 'on' despite not showing it), it could have caused the item loss I saw.

**Comment by Meatnog** [comment]
Ok, we found it!  At least in our setup.   What we found was sometimes the ME Export Bus doesn't properly work with a redstone signal.     So if you have an ME Level Bus aimed at an ME Export Bus and the ME Export Bus is set to only work with a signal, sometimes it either doesn't recognize the signal as being on, and sometimes it doesn't recognize it going OFF.    It is very, very infrequent.   But it is there.   I do not know if this is a TT issue or AE issue.

**Comment by geoffcorey** [comment]
Running TT on Utlimate 1.0.1 with AE upgraded to r.9.i
AE system spanning 2 chunks
- chunk 1
  -- ME Controller/ME Drive/Terminal
  -- Fusion reactor cell exports
  -- 2 Ender Chests with ME Storage Bus (stocked with ender eye dust, empty cells)
  -- 1 Ender Chest with ME Import Bus (importing 40K aluminium dust, single)
  -- Industrial blaster (making aluminium ingots continuously) w/ME import and export bus
  -- Matfab  w/ME import and export bus
- chunk 2
  -- 3 more  Industrial blasters (making aluminium ingots continuously) w/ME import and export bus
  -- 1 singularity compressor (making aluminium blocks continuously) w/ME import and export bus

It has been running continuously for 24 hours and no item loss

**Comment by HaoSs07** [comment]
hopefully i will be able to test r9i this week also

**Comment by HaoSs07** [comment]
From the first round of tests it seems no more losing items :) using the same setup,(is on a multiple chunks) with no preformated disks 
FTB 1.1.2

**Comment by HaoSs07** [comment]
more heavy testing, 0 items missing :)

**Comment by LunNova** [comment]
> `<Kenneaal> Aye, we've been doing testing too. The only issue we've seen is with non-formatted disks in ME chests connected to ME networks.`

Anyone able to confirm that this is no longer an issue with updated AE?

**Comment by Meatnog** [comment]
Nallar, I am happy to say it is no longer an issue.   We have discovered an issue with ME Export Busses getting 'stuck' on/off (when they're set to receive a redstone signal), but I believe that's an AE issue and I opened it with AlgorithmX.

**Comment by LunNova** [comment]
:)

**Comment by HaoSs07** [comment]
well someone is happy :D

**Comment by nylser** [comment]
I'm also very happy this issue is finally resolved, so we are no longer stuck there
